pallets/validators-manager/src/lib.rs

```diff
@@ -248,25 +250,21 @@ pub mod pallet {
         pub fn add_collator(
             origin: OriginFor<T>,
             collator_account_id: T::AccountId,
-            validator_eth_public_key: ecdsa::Public,
+            collator_eth_public_key: ecdsa::Public,
             deposit: Option<BalanceOf<T>>,
         ) -> DispatchResultWithPostInfo {
             ensure_root(origin)?;
             let validator_account_ids =
                 Self::validator_account_ids().or_else(|| Some(vec![])).expect("empty vec");
             ensure!(validator_account_ids.len() > 0, Error::<T>::NoValidators);
-            ensure!(
-                !<EthereumPublicKeys<T>>::contains_key(&validator_eth_public_key),
-                Error::<T>::ValidatorEthKeyAlreadyExists
-            );
 
             ensure!(
                 !validator_account_ids.contains(&collator_account_id),
                 Error::<T>::ValidatorAlreadyExists
             );
             ensure!(
-                !<EthereumPublicKeys<T>>::contains_key(&validator_eth_public_key),
-                Error::<T>::ValidatorAlreadyExists
+                !<EthereumPublicKeys<T>>::contains_key(&collator_eth_public_key),
+                Error::<T>::ValidatorEthKeyAlreadyExists
```

Comment:
Errors are enumerations and in the past, the append-only rule applied to them as it could break the way the client decodes things if you alter them.
Not sure if it is still applicable since the clients are querying the runtime for the chain-specific types.